app/models/container_node.rb

```diff
@@ -75,6 +75,12 @@ def ipaddress
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    if MiqRegion.my_region.role_active?("cockpit_ws")
+      uri = URI("/dashboard/cockpit_redirect")
+      uri.query = {:ip => ipaddress }.to_query
+    else
+      uri = URI::HTTP.build(:host => ipaddress, :port => 9090)
```

Comment:
Right, I see what you're saying.  So, if they wanted to use a normal cockpit without an embedded proxying worker, they'd need to open this port.  Ok